scripts/json2src.py

```diff
@@ -0,0 +1,58 @@
+#! /usr/bin/env python3
+"""
+ Copyright (C) 2019-2021 Intel Corporation
+
+ SPDX-License-Identifier: MIT
+
+"""
+import argparse
+import util
+import generate_code
+import os, sys
+import time
+import json
+
+"""
+    helper for adding mutually-exclusive boolean arguments "--name" and "--skip-name"
+"""
+def add_argument(parser, name, help, default=False):
+    group = parser.add_mutually_exclusive_group(required=False)
+    group.add_argument("--" + name, dest=name, help="Enable "+help, action="store_true")
+    group.add_argument("--skip-" + name, dest=name, help="Skip "+help, action="store_false")
+    parser.set_defaults(**{name:default})
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    add_argument(parser, "lib", "generation of lib files.", True)
+    add_argument(parser, "loader", "generation of loader files.", True)
+    add_argument(parser, "layers", "generation of layer files.", False)
+    add_argument(parser, "drivers", "generation of null driver files.", True)
+    parser.add_argument("--debug", action='store_true', help="dump intermediate data to disk.")
+    parser.add_argument("--sections", type=list, default=None, help="Optional list of sections for which to generate source, default is all")
+    parser.add_argument("--ver", type=str, default="1.0", help="specification version to generate.")
+    parser.add_argument('--api-json', nargs='?', type=argparse.FileType('r'), default=sys.stdin, help="JSON file containing the API specification, by default read from stdin")
+    parser.add_argument("out_dir", type=str, help="Root of the loader repository.")
+    args = parser.parse_args()
+
+    input = json.loads(args.api_json.read())
+
+    start = time.time()
+
+    srcpath = os.path.join(args.out_dir, "source")
+
+    for idx, specs in enumerate(input['specs']):
+        config = input['configs'][idx]
+        if args.sections == None or config['name'] in args.sections:
+            if args.lib:
+                generate_code.generate_lib(srcpath, config['name'], config['namespace'], config['tags'], args.ver, specs, input['meta'])
+            if args.loader:
+                generate_code.generate_loader(srcpath, config['name'], config['namespace'], config['tags'], args.ver, specs, input['meta'])
+            if args.layers:
+                generate_code.generate_layers(srcpath, config['name'], config['namespace'], config['tags'], args.ver, specs, input['meta'])
```

Comment:
I don't want to remove it entirely since we will be adding the layer functionality soon. I commented out these lines, so it's impossible to call this a non-existent method.